src/main/java/org/spongepowered/common/registry/type/entity/GameModeRegistryModule.java

```diff
@@ -47,7 +47,7 @@ protected void generateInitialMap() {
 
     public static GameType toGameType(GameMode gamemode) {
         for (GameType gameType : GameType.values()) {
-            if (gameType.getName().equalsIgnoreCase(gamemode.getId())) {
+            if (gameType.getName().equals(gamemode.getName())) {
```

Comment:
Good spot.

Given that we guarantee IDs to be unique, and not names, should this not perform an equality check on the ID instead?